src/ImageLoaders/IHex32/IntelHex32Loader.cs

```diff
@@ -0,0 +1,242 @@
+                BaseAddress = bAddr;
+                Datum = new List<byte>();
+            }
+
+            public MemChunk(uint address) : this(Address.Ptr32(address))
+            {
+            }
+
+            public bool Contains(Address addr) => (BaseAddress <= addr && EndAddress > addr);
+
+            public bool IsAtTail(Address addr) => addr == EndAddress;
+
+            public bool IsAtHeadOf(Address addr) => EndAddress == addr;
+
+        }
+
+        /// <summary>
+        /// List of memory chunks.
+        /// </summary>
+        internal class MemoryChunksList : IEnumerable<MemChunk>
+        {
+            private Address currAddr = null;
+            private Address nextAddr = null;
+            private SortedList<Address, MemChunk> memChunks;
+            private MemChunk currMemChunk;
+
+            public MemoryChunksList()
+            {
+                memChunks = new SortedList<Address, MemChunk>();
+                currMemChunk = null;
+            }
+
+            private MemChunk GetPredChunk(Address addr)
+                => memChunks.Values.FirstOrDefault(mchk => mchk.IsAtHeadOf(addr));
+
+            /// <summary>
+            /// Adds a series of data byte to the memory chunks list as decoded from an IHex32 data record.
+            /// </summary>
+            /// <remarks>
+            /// The current implementation assumes that each IHex32 data record loads unique data block (no overlap).
+            /// </remarks>
+            /// <param name="address">The address from the decoded IHex32 record.</param>
+            /// <param name="data">The binary bytes contained in the IHex32 record.</param>
+            public void AddData(uint address, byte[] data)
+            {
+                if (data == null || data.Length < 1)
+                    return;
+
+                if (currAddr == null)
+                {
+                    currAddr = Address.Ptr32(address);
+                    currMemChunk = new MemChunk(currAddr);
+                    memChunks.Add(currAddr, currMemChunk);
+                }
+                else
+                {
+                    currAddr = Address.Ptr32(address);
+                    if (nextAddr != currAddr)
+                    {
+                        currMemChunk = GetPredChunk(currAddr);
+                        if (currMemChunk == null)
+                        {
+                            currMemChunk = new MemChunk(currAddr);
+                            memChunks.Add(currAddr, currMemChunk);
+                        }
+                    }
+                }
+
+                currMemChunk.Datum.AddRange(data);
+                nextAddr = currAddr + data.Length;
+            }
+
+            public IEnumerator<MemChunk> GetEnumerator() => memChunks.Values.GetEnumerator();
+
+            IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+        }
+
+        #endregion
+
+        #region Locals
+
+        private DecompilerEventListener listener;
```

Comment:
No `#region`, and fields should be towards the top of the class. See the style guide.